lib/uri_checker.rb

```diff
@@ -0,0 +1,213 @@
+require "uri"
+
+module UriChecker
+  class Report
+    attr_reader :errors, :warnings
+
+    def initialize(errors: nil, warnings: nil)
+      @errors = errors || Hash.new { |hash, key| hash[key] = [] }
+      @warnings = warnings || Hash.new { |hash, key| hash[key] = [] }
+    end
+
+    def merge!(other)
+      errors.merge!(other.errors) { |key, oldval, newval| oldval | newval }
+      warnings.merge!(other.warnings) { |key, oldval, newval| oldval | newval }
+    end
+
+    def has_errors?
+      !errors.empty?
+    end
+  end
+
+  class ValidUri
+    HTTP_URI_SCHEMES = %w(http https)
```

Comment:
Not a concern for this PR but would using a combination of `config_for` and `.yml` files help managing the many config options in these Checker classes?